src/rfc3986/_mixin.py

```diff
@@ -56,6 +56,15 @@ def authority_info(self):
     def _match_subauthority(self):
         return misc.SUBAUTHORITY_MATCHER.match(self.authority)
 
+    def _get_base_uri_validator(self):
```

Comment:
I'd rather this be a property that looks like:

```py
@property 
def _validator(self):
    v = getattr(self, '_cached_validator', None)
    if v is not None:
        return v
    self._cached_validator = validators.Validator().require_presence_of("scheme")
    return self._cached_validator
```

And then used as 

```
self._validator.validate(base_uri)
```

Reply:
@sigmavirus24 thanks, I've updated this and added a test needed to cover the case of the validator already existing with this code arrangement.